networkit/cpp/io/NetworkitBinaryReader.cpp

```diff
@@ -89,12 +116,15 @@ Graph NetworkitBinaryReader::read(const std::string& path) {
         firstVert.push_back(first);
     }
     firstVert.push_back(nodes);
-
     // Read adjacency lists.
-    const char *adjIt = mmfile.cbegin() + header.offsetAdjLists;
-    const char *transpIt = mmfile.cbegin() + header.offsetAdjTranspose;
-    const char *adjWghtIt = mmfile.cbegin() + header.offsetWeightLists;
-    const char *transpWghtIt = mmfile.cbegin() + header.offsetWeightTranspose;
+    const char *adjIt = getIterator(source) + header.offsetAdjLists;
+    const char *transpIt = getIterator(source) + header.offsetAdjTranspose;
+    const char *adjWghtIt = getIterator(source) + header.offsetWeightLists;
+    const char *transpWghtIt = getIterator(source) + header.offsetWeightTranspose;
+    const char *adjIdIt = getIterator(source) + header.offsetAdjIdLists;
+    const char *transpIdIt = getIterator(source) + header.offsetAdjIdTranspose;
```

Comment:
Save the result of `getIterator(source)` to a local variable to avoid duplication here.